index.d.ts

```diff
@@ -83,15 +83,25 @@ export interface SelectQuery extends BaseQuery {
 
 export interface PathsQuery extends BaseQuery {
     queryType: "PATHS";
-    start: IriTerm ;
-    via: IriTerm ;
-    end: IriTerm ;
+    start: PathValue;
+    via: PathViaValue;
+    end: PathValue;
+    all?: boolean|undefined;
     maxlength?: number| undefined;
     shortest?: boolean | undefined;
     limit?: number | undefined;
     offset?: number | undefined;
     cyclic?: boolean | undefined;
 }
+export interface PathValue {
+    var: Var;
+    value: IriTerm | GraphPattern;
+}
+
+export interface PathViaValue {
+    var: Var | null;
+    value: IriTerm | Var | GraphPattern;
```

Comment:
```suggestion
    value: Var | PropertyPath | Pattern[];
```

Reply:
I changed `GraphPattern` to `Pattern[]`, based on looking at what was used for `WHERE`. Pretty sure `GraphPattern` is for specifying a named graph.